style-guide.rst

```diff
@@ -46,10 +44,8 @@ Headings
           Heading
           .......
      - H5 heading
-   * - Underlines with other characters, for example, ``"``, ``#`` or ``*``
-     - Further headings
 
-Underlines and overlines must be at least as long as the title or heading.
+Underlines must be the same length as the title or heading.
```

Comment:
I think this is overly prescriptive; please let's keep "must" separate from in-house convention. The "musts" are:

"Section headers (ref) are created by underlining (and optionally overlining) the section title with a punctuation character, [at least as long as the text.](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections)"

and, by following the "(ref)" from there:

"[When an overline is used, the length and character used must match the underline.](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#sections)"

It's easy to change a heading if its under- and over-lines are made too long deliberately. If instead their length must match that of the heading, they have to be dressed with almost every change. That makes for pretty reST, but it's error-prone and doesn't affect what Sphinx generates.

Reply:
But this is supposed to be a reST style guide, so it is all about "pretty reST" and not only stating the requirements of reST (those are linked).

So if we want the underlines to be the same length, we should state it here. 
Do we want it? 
I vote yes, and I think @pmatulis also did, and @lblythen votes no.
Other votes?

Reply:
Yes, I do think we need to be opinionated here and also maintain this approach throughout the guide. Indeed, this is not a reference guide but a style guide. Otherwise the guide will

a) become unnecessarily wordy by having to explain "must" vs "should" vs "may" instead of just "do this"
b) lead to a variety of styles throughout company projects over time - the present case is just one example